File: 프로그래머스/lv2/76502. 괄호 회전하기/괄호 회전하기.java
import java.util.Stack;
class Solution {
    public int solution(String s) {
        int answer = 0;
        int len = s.length();
        int x =0;

       Stack<Character> ck = new Stack<>();
        while(x!=len){
            ck.clear();
            for(int j=0; j<len; j++){
                Character target = s.charAt(j);
                if(ck.isEmpty()){
                    ck.push(target);
                }else{
                    if(target == ']'){
                        if(ck.peek() == '['){
                            ck.pop();
                        }
                        else{
                            ck.push(target);
                        }
                    } 
                    else if(target == ')'){
                        if(ck.peek() == '('){
                            ck.pop();
                        }
                        else{
                            ck.push(target);
                        }
                    } 
                    else if(target == '}'){
                        if(ck.peek() == '{'){
                            ck.pop();
                        }
                        else{
                            ck.push(target);
                        }
                    }
                    else{
                        ck.push(target);
                    }
                }
            }   
            if(ck.isEmpty()){
                answer++;
            }
            s = s.substring(1, len) + s.split("")[0];
            x++;
            
        }
        return answer;
    }
}

File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.js
function solution(s) {
    let answer = 0;
    
    let stack = [];
  
    if (s.length % 2 === 1) return 0;
    
    for(let i = 0; i < s.length; i++){       
        let str = s.slice(i) + s.slice(0,i);
        let isPossible = true;
      
        for(const n of str){
            if(n === "[" || n === "{" || n === "(" ){
                stack.push(n);
            }else{
                let pop = stack.pop();
                
                if (pop === "(" && n === ")") continue;
                if (pop === "{" && n === "}") continue;
                if (pop === "[" && n === "]") continue;

                isPossible = false;
                break;
            };
        };
        if (isPossible) answer++;
    };
    return answer;
};